### PR TITLE
Rework GetCourses in CoursesController

### LINKED ISSUE #304: Add new endpoint for CoursesController

### PR DESCRIPTION
**Pull request by QuestroL**

### CODE REVIEW

File: CharlieBackend.Data/Repositories/Impl/CourseRepository.cs [hunk 1]
```diff
@@ -37,6 +37,14 @@ public async Task<bool> IsCourseHasGroupAsync(long id)
             return await _applicationContext.StudentGroups.AnyAsync(s => s.CourseId == id);
         }
 
+        public async Task<IList<Course>> GetAllActiveCourses()
```
Comment (filinskyi) [inline CharlieBackend.Data/Repositories/Impl/CourseRepository.cs]: due to the naming convention please provide ”Async” suffix to the end of the method name.

File: CharlieBackend.Data/Repositories/Impl/CourseRepository.cs [hunk 2]
```diff
@@ -37,6 +37,14 @@ public async Task<bool> IsCourseHasGroupAsync(long id)
...
+        {
+            return await _applicationContext.Courses
+                .Where(x => x.IsActive == true)
+                .Select(x => x)
```
Comment (filinskyi) [inline CharlieBackend.Data/Repositories/Impl/CourseRepository.cs]: this query can be simplified, due to type of the parameter isActive is bool we can use it as a predicate for the Where method, and also Select part is don’t needed.
Code: .Where(x => x.IsActive)

**Review by filinskyi — CHANGES_REQUESTED** [review]
a query simplification and method naming

**Review by filinskyi — APPROVED** [review]

File: CharlieBackend.Api/Controllers/CoursesController.cs [hunk 3]
```diff
@@ -54,6 +54,19 @@ public async Task<IList<CourseDto>> GetAllCourses()
             return courses;
         }
 
+        /// <summary>
+        /// Gets all active cources
+        /// </summary>
+        /// <response code="200">Successful return of list of all active courses</response>
+        [Authorize(Roles = "Admin, Mentor, Secretary, Student")]
```
Comment (Hlib-mats) [inline CharlieBackend.Api/Controllers/CoursesController.cs]: What abouat swagger examples?

**Review by Hlib-mats — APPROVED** [review]

File: CharlieBackend.Api/Controllers/CoursesController.cs [hunk 4]
```diff
@@ -42,14 +42,14 @@ public async Task<ActionResult<CreateCourseDto>> PostCourse(CreateCourseDto cour
...
 
         /// <summary>
-        /// Gets all cources
+        /// Gets cources
         /// </summary>
         /// <response code="200">Successful return of list of courses</response>
         [Authorize(Roles = "Admin, Mentor, Secretary, Student")]
-        [HttpGet]
-        public async Task<IList<CourseDto>> GetAllCourses()
+        [HttpGet("isAvtive")]
```
Comment (gingerbread-monster) [inline CharlieBackend.Api/Controllers/CoursesController.cs]: typo
```suggestion
        [HttpGet("isActive")]
```

File: CharlieBackend.Api/Controllers/CoursesController.cs [hunk 5]
```diff
@@ -42,14 +42,14 @@ public async Task<ActionResult<CreateCourseDto>> PostCourse(CreateCourseDto cour
         }
 
         /// <summary>
-        /// Gets all cources
+        /// Gets cources
```
Comment (gingerbread-monster) [inline CharlieBackend.Api/Controllers/CoursesController.cs]:
```suggestion
        /// Gets courses
```

File: CharlieBackend.Api/Controllers/CoursesController.cs [hunk 6]
```diff
@@ -42,14 +42,14 @@ public async Task<ActionResult<CreateCourseDto>> PostCourse(CreateCourseDto cour
...
         /// </summary>
         /// <response code="200">Successful return of list of courses</response>
         [Authorize(Roles = "Admin, Mentor, Secretary, Student")]
-        [HttpGet]
-        public async Task<IList<CourseDto>> GetAllCourses()
+        [HttpGet("isAvtive")]
+        public async Task<IList<CourseDto>> GetCourses(bool? isActive)
```
Comment (gingerbread-monster) [inline CharlieBackend.Api/Controllers/CoursesController.cs:54]: * `activeOnly` should be a more suitable and descriptive name for this argument
* please add parameter description:
```cs
/// <param name="activeOnly">description</param>
```

**Review by gingerbread-monster — APPROVED** [review]

File: CharlieBackend.Api/Controllers/CoursesController.cs [hunk 7]
```diff
@@ -42,14 +42,18 @@ public async Task<ActionResult<CreateCourseDto>> PostCourse(CreateCourseDto cour
         }
 
         /// <summary>
-        /// Gets all cources
+        /// Gets courses
         /// </summary>
+        /// <param name="isActive">
+        /// 1. If IsActive true method return all active Courses.
+        /// 2. If IsActive false method return all not active Courses.
+        /// 3. If IsActive null method return all courses. </param>
```
Comment (gingerbread-monster) [inline CharlieBackend.Api/Controllers/CoursesController.cs]: Minor grammar corrections
```suggestion
        /// 1. If IsActive is true – endpoint returns all active Courses.
        /// 2. If IsActive is false – endpoint returns all not active Courses.
        /// 3. If IsActive is null – endpoint returns all courses. </param>
```

**Pull request merged**